src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Simplification/Simplifiers/CastSimplifier.cs

```diff
@@ -768,7 +775,7 @@ private static bool CastRemovalWouldCauseSignExtensionWarning(ExpressionSyntax e
             var leftOperation = semanticModel.GetOperation(leftOperand.WalkDownParentheses(), cancellationToken);
             var rightOperation = semanticModel.GetOperation(rightOperand.WalkDownParentheses(), cancellationToken);
 
-            if (leftOperand == null || rightOperand == null)
+            if (leftOperation == null || rightOperation == null)
```

Comment:
⚠️ @CyrusNajmabadi As best I can tell, this is the null check we always wanted anyway

Reply:
👍  yup yup!